### PR TITLE
Move OperationsEvents to EventCore

### PR DESCRIPTION
**Pull request by kagemand-andersen**

### CODE REVIEW

File: src/main/java/org/dcsa/core/events/service/impl/OperationsEventTOServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Service
+public class OperationsEventTOServiceImpl extends BaseServiceImpl<OperationsEventTO, UUID> implements OperationsEventTOService {
```
Comment (HenrikHL) [inline src/main/java/org/dcsa/core/events/service/impl/OperationsEventTOServiceImpl.java:22]: Is this class needed? I thought you said we don't need it?

Comment (kagemand-andersen) [reply src/main/java/org/dcsa/core/events/service/impl/OperationsEventTOServiceImpl.java:22]: I don't know if OperationsEventTO is used or something we should have.

File: src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    private String changeRemark;
+
+    @Column("delay_reason_code")
+    private String delayReasonCode;
+
+    @Column("vessel_position")
+    private String vesselPositionID;
+
+    @Transient
+    private TransportCallTO transportCall;
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java:61]: Shouldn't this too go into `OperationsEventTO`?

File: src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Transient
+    private LocationTO vesselPosition;
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java:64]: Ditto  as above?

File: src/main/java/org/dcsa/core/events/service/OperationsEventService.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.dcsa.core.events.service;
+
+import org.dcsa.core.events.model.OperationsEvent;
+import org.dcsa.core.service.ExtendedBaseService;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+public interface OperationsEventService extends ExtendedBaseService<OperationsEvent, UUID> {
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/core/events/service/OperationsEventService.java:9]: Extend `EventService<OperationsEvent>` instead.

File: src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java (hunk 2 above)
Comment (kagemand-andersen) [reply src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java:61]: Do you mean `LocationTO` and `TransportCallTO` should be moved from `AbstractOperationsEvent` to `OperationsEventTO`?

File: src/main/java/org/dcsa/core/events/service/impl/GenericEventServiceImpl.java [hunk 5]
```diff
@@ -44,51 +30,54 @@ public EventRepository getRepository() {
...
-                    return transportEventService.loadRelatedEntities((TransportEvent) event);
-                case EQUIPMENT:
-                    return equipmentEventService.loadRelatedEntities((EquipmentEvent) event);
-                case SHIPMENT:
-                    return shipmentEventService.loadRelatedEntities((ShipmentEvent)event);
-                default:
-                    return Mono.just(event);
-            }
-        });
+        throw new NotImplementedException();
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/core/events/service/impl/GenericEventServiceImpl.java:33]: This change will affect TNT, any particular reason for the change?

File: src/main/java/org/dcsa/core/events/service/impl/GenericEventServiceImpl.java [hunk 6]
```diff
@@ -44,51 +30,54 @@ public EventRepository getRepository() {
...
     }
 
-  @Override
-  public Mono<Event> findById(UUID id) {
-    return Mono.<Event>empty()
-        .switchIfEmpty(
-            transportEventService
+    @Override
+    public Mono<Event> findById(UUID id) {
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/core/events/service/impl/GenericEventServiceImpl.java:37]: Same here, It will effect TNT.

**Review by douradofrazer — CHANGES_REQUESTED** [review]

File: src/main/java/org/dcsa/core/events/service/impl/GenericEventServiceImpl.java (hunk 5 above)
Comment (kagemand-andersen) [reply src/main/java/org/dcsa/core/events/service/impl/GenericEventServiceImpl.java:33]: It is to ensure that generic isn't just the TNT implementation as it was previously.

There is a PR for this: https://github.com/dcsaorg/DCSA-TNT/pull/56

File: src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java (hunk 2 above)
Comment (kagemand-andersen) [reply src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java]: I have moved them out into `OperationsEventTO`.

File: src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java (hunk 3 above)
Comment (kagemand-andersen) [reply src/main/java/org/dcsa/core/events/model/base/AbstractOperationsEvent.java]: Moved to `OperationsEventTO`.

**Review by HenrikHL — APPROVED** [review]

**Review by douradofrazer — APPROVED** [review]

**Pull request merged**